From b45c9a58b73f654b74518119ffc0198987dbc86a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:10 -0400
Subject: [tools] include: Move perf's bug.h to a generic place

Message-id: <1408953210-26343-243-git-send-email-jolsa@redhat.com>
Patchwork-id: 89000
O-Subject: [PATCH RHEL7.1 BZ1133083 242/322] tools include: Move perf's bug.h to a generic place
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 02dfc8d775f0709ab494d4b2cce12c8429ff7530
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 9 23:00:54 2014 +0900

description
===========
So that it can be shared with others like libtraceevent.

[ Added the new header to tools/perf/MANIFEST ]
---
 create mode 100644 tools/include/asm/bug.h
 delete mode 100644 tools/perf/util/include/asm/bug.h

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/include/asm/bug.h b/tools/include/asm/bug.h
new file mode 100644
index 0000000..eca78df
--- /dev/null
+++ b/tools/include/asm/bug.h
@@ -0,0 +1,23 @@
+#ifndef _TOOLS_ASM_BUG_H
+#define _TOOLS_ASM_BUG_H
+
+#define __WARN_printf(arg...) do { fprintf(stderr, arg); } while (0)
+
+#define WARN(condition, format...) ({  \
+ int __ret_warn_on = !!(condition); \
+ if (unlikely(__ret_warn_on))  \
+  __WARN_printf(format);  \
+ unlikely(__ret_warn_on);  \
+})
+
+#define WARN_ONCE(condition, format...) ({ \
+ static int __warned;   \
+ int __ret_warn_once = !!(condition); \
+      \
+ if (unlikely(__ret_warn_once))  \
+  if (WARN(!__warned, format))  \
+   __warned = 1;  \
+ unlikely(__ret_warn_once);  \
+})
+
+#endif /* _TOOLS_ASM_BUG_H */
diff --git a/tools/perf/MANIFEST b/tools/perf/MANIFEST
index 285f28f..4e53535 100644
--- a/tools/perf/MANIFEST
+++ b/tools/perf/MANIFEST
@@ -4,6 +4,7 @@ tools/lib/traceevent
 tools/lib/lk
 tools/lib/symbol/kallsyms.c
 tools/lib/symbol/kallsyms.h
+tools/include/asm/bug.h
 tools/include/linux/compiler.h
 include/linux/const.h
 include/linux/perf_event.h
diff --git a/tools/perf/Makefile.perf b/tools/perf/Makefile.perf
index 6be0676..87d7726 100644
--- a/tools/perf/Makefile.perf
+++ b/tools/perf/Makefile.perf
@@ -226,7 +226,7 @@ LIB_H += util/include/linux/string.h
 LIB_H += util/include/linux/types.h
 LIB_H += util/include/linux/linkage.h
 LIB_H += util/include/asm/asm-offsets.h
-LIB_H += util/include/asm/bug.h
+LIB_H += ../include/asm/bug.h
 LIB_H += util/include/asm/byteorder.h
 LIB_H += util/include/asm/hweight.h
 LIB_H += util/include/asm/swab.h
diff --git a/tools/perf/util/include/asm/bug.h b/tools/perf/util/include/asm/bug.h
deleted file mode 100644
index 7fcc681..0000000
--- a/tools/perf/util/include/asm/bug.h
+++ /dev/null
@@ -1,22 +0,0 @@
-#ifndef _PERF_ASM_GENERIC_BUG_H
-#define _PERF_ASM_GENERIC_BUG_H
-
-#define __WARN_printf(arg...) do { fprintf(stderr, arg); } while (0)
-
-#define WARN(condition, format...) ({  \
- int __ret_warn_on = !!(condition); \
- if (unlikely(__ret_warn_on))  \
-  __WARN_printf(format);  \
- unlikely(__ret_warn_on);  \
-})
-
-#define WARN_ONCE(condition, format...) ({ \
- static int __warned;   \
- int __ret_warn_once = !!(condition); \
-      \
- if (unlikely(__ret_warn_once))  \
-  if (WARN(!__warned, format))  \
-   __warned = 1;  \
- unlikely(__ret_warn_once);  \
-})
-#endif
-- 
1.7.1